akka-cluster/src/multi-jvm/scala/akka/cluster/MultiDcSplitBrainSpec.scala

```diff
@@ -175,13 +188,104 @@ abstract class MultiDcSplitBrainSpec
       }
       enterBarrier("node-4-left")
 
-      unsplitDataCenters(notMembers = Set(fourth))
+      unsplitDataCenters(notMembers = Set(fourth, fifth))
 
       runOn(first, second) {
         awaitAssert(clusterView.members.filter(_.address == address(fourth)) should ===(Set.empty))
       }
       enterBarrier("inter-data-center-split-2-done")
     }
 
+    "be able to have data center member restart (same host:port) while there is inter data center split" in within(40.seconds) {
+      val subscribeProbe = TestProbe()
+      runOn(first, second, third, fifth) {
+        Cluster(system).subscribe(subscribeProbe.ref, InitialStateAsSnapshot, classOf[MemberUp], classOf[MemberRemoved])
+        subscribeProbe.expectMsgType[CurrentClusterState]
+      }
+      enterBarrier("subscribed")
+      runOn(fifth) {
+        Cluster(system).join(third)
+      }
+      var fifthOriginalUniqueAddress: Option[UniqueAddress] = None
+      runOn(first, second, third, fifth) {
+        awaitAssert(clusterView.members.collect {
+          case m if m.dataCenter == "dc2" && m.status == MemberStatus.Up ⇒ m.address
+        } should ===(Set(address(third), address(fifth))))
+        fifthOriginalUniqueAddress = clusterView.members.collectFirst { case m if m.address == address(fifth) ⇒ m.uniqueAddress }
+      }
+      enterBarrier("fifth-joined")
+
+      splitDataCenters(notMembers = Set(fourth))
+
+      runOn(fifth) {
+        Cluster(system).shutdown()
+      }
+      runOn(third) {
+        awaitAssert(clusterView.members.collect {
+          case m if m.dataCenter == "dc2" ⇒ m.address
+        } should ===(Set(address(third))))
+      }
+      enterBarrier("fifth-removed")
+
+      runOn(fifth) {
+        // we can't use any multi-jvm test facilities on fifth after this, because have to shutdown
+        // actor system to be able to start new with same port
+        val thirdAddress = address(third)
+        enterBarrier("fifth-waiting-for-termination")
+        Await.ready(system.whenTerminated, remaining)
+
+        val restartedSystem = ActorSystem(
+          system.name,
+          ConfigFactory.parseString(s"""
+            akka.remote.netty.tcp.port = ${Cluster(system).selfAddress.port.get}
+            akka.remote.artery.canonical.port = ${Cluster(system).selfAddress.port.get}
+            akka.coordinated-shutdown.terminate-actor-system = on
+            """).withFallback(system.settings.config))
+        Cluster(restartedSystem).join(thirdAddress)
+        Await.ready(restartedSystem.whenTerminated, remaining)
+      }
+
+      // multi-jvm test facilities on fifth after this
```

Comment:
Still not any multi-jvm-facilities on fifth, right? Or am I reading this comment wrong?

Reply:
yeah, missing the "no" word